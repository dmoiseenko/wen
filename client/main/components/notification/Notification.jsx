import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';


function Notification({
  description,
  remove,
  id,
  isError,
  isEvent
}) {
  function handleClick() {
    remove(id);
  }

  const notificationClassName = classNames(
    'notification',
    { 'is-warning': isError },
    { 'is-primary': isEvent }
  );

  return (
    <div
      className={notificationClassName}
      onClick={handleClick}
    >
      {description}
    </div>
  );
}

Notification.propTypes = {
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
  isError: PropTypes.bool.isRequired,
  isEvent: PropTypes.bool.isRequired
};

export default Notification;
