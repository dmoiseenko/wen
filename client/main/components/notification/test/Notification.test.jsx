import React from 'react';
import { shallow } from 'enzyme';

import Notification from '../Notification';


const props = {
  description: 'notification description',
  id: '1',
  remove: jest.fn()
};


it('should render event notification', () => {
  const wrapper = shallow(
    <Notification
      {...props}
      isEvent
      isError={false}
    />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should render error notification', () => {
  const wrapper = shallow(
    <Notification
      {...props}
      isError
      isEvent={false}
    />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call remove when click fires', () => {
  const remove = jest.fn();

  const wrapper = shallow(
    <Notification
      {...props}
      remove={remove}
      isError
      isEvent={false}
    />
  );

  wrapper.find('.notification').simulate('click');

  expect(remove).toHaveBeenCalledWith('1');
});
