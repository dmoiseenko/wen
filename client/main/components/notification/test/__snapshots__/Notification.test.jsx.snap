// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render error notification 1`] = `
<div
  className="notification is-warning"
  onClick={[Function]}
>
  notification description
</div>
`;

exports[`should render event notification 1`] = `
<div
  className="notification is-primary"
  onClick={[Function]}
>
  notification description
</div>
`;
